src/MonitoringDemo/ProcessGroup.cs

```diff
@@ -216,9 +216,9 @@ private void DisposeWindowsJob()
             RedirectStandardOutput = true
         };
 
-        if (arguments is not null)
+        foreach (var a in arguments)
```

Comment:
Doesn't this require something like ?? Enumerable.Empty<string> because the arguments array could be null?